r/R/dataset-format.R

```diff
@@ -122,6 +122,18 @@ CsvFileFormat$create <- function(...,
                                  opts = csv_file_format_parse_options(...),
                                  convert_options = csv_file_format_convert_opts(...),
                                  read_options = csv_file_format_read_opts(...)) {
+
+  options <- list(...)
+  schema  <- options[["schema"]]
+
+  if (length(read_options$column_names) > 0 & !is.null(schema) & !identical(names(schema), read_options$column_names)) {
+    abort(c(
+        '"column_names" in read_options do not match the schema.',
+      i = "Set column_names in read_options to match the schema",
+      i = "Omit the read_options argument"
+    ))
```

Comment:
Looking good, just a couple of suggestions:
- the user of the `open_dataset()` function won't necessarily have used the function `read_options()` directly, so we should remove the reference to it to avoid confusion
- to provide maximal useful information to the end-user, could you print out the arguments which are mismatches between `column_names` and the schema?

Reply:
Also, please could you add a test which triggers this error?

Reply:
Thanks for the good comments!

It raises an error and prints out the mismatches between read_options$column_names and names(schema).

At the moment a mismatch is either due to 1) set difference or a 2) different order of column names. However, I'm a bit unsure of the latter. Do we in fact want to raise an error if the order of names differ? For example: `c("a","b") != c("a","b")`

Also, if this is the case, I'd be interested if you have any tips on convenient ways to do this. I resorted to using `suppressWarnings` to handle cases where vector lengths differ.